setup/tue-functions.bash

```diff
@@ -19,6 +19,166 @@ function _list_subdirs
     done
 }
 
+# ----------------------------------------------------------------------------------------------------
+#                                       GIT LOCAL HOUSEKEEPING
+# ----------------------------------------------------------------------------------------------------
+
+function __tue-default-branch
+{
+    local default_branch
+    default_branch=$(git remote show origin | grep HEAD | awk '{print $3}')
+    _git_remote_checkout origin "$default_branch"
+}
+
+function _tue-default-branch
+{
+    _tue-repos-do "__tue-default-branch"
+}
+
+function _tue-git-branch-clean
+{
+    # Function to remove stale branches from a git repository (which should
+    # either be the PWD or one of its parent directories). The function removes
+    # stale branches in two layers. First it removes all branches that have been
+    # merged in the remote, then it checks for unmerged branches that have been
+    # deleted from the remote and prompts for confirmation before removal. If
+    # the function is called with "--force-remove" flag, then no confirmation is asked
+
+    local force_remove=
+    local error_code=
+    local stale_branches=
+    local repo=
+    repo="$PWD"
```

Comment:
I think it is if you use `_tue-repos-do` because that command first does `cd` into the directory and then evaluate the string command.

Taking that into account, I have written this function in a way that it is executed only inside a git repo. If you look at the statement of `git fetch -p`, then you will see that the function exits if `git fetch -p` fails (one of the reason also includes that the directory is not a git repo)